src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java

```diff
@@ -115,9 +115,19 @@ public void clearSelection() {
         setSelection(Selection.UNDEFINED);
     }
 
+    /**
+     * Inserts new text into the document at current caret position
+     * Smart enough to distinguish between append and insert operations
+     * @param text text to insert
+     */
     void insert( String text ) {
         removeSelection();
-        textBuffer.insert(text, getCaretPosition());
+        int caretPosition =  getCaretPosition();
+        if ( caretPosition >= getTextLength()) {
+            textBuffer.append(text);
+        } else {
+            textBuffer.insert(text, caretPosition);
+        }
```

Comment:
I was already wondering if the next line should move caret position the length of the text?

Reply:
Good call... currently we move the caret to 1 char, for some reason. I think it has to move by `text.length()`.
<img width="475" alt="image" src="https://user-images.githubusercontent.com/341840/153874925-9aa2d2e0-da4b-4a67-a773-15d6e770e177.png">

Reply:
Done